src/lib/Microsoft.Health.Fhir.Ingest/Service/EventProcessingMeter.cs

```diff
@@ -19,21 +20,23 @@ public Task<EventStats> CalculateEventStats(IEnumerable<EventData> events)
 
             foreach (var e in events)
             {
-                var bodySizeBytes = e.Body.Array.Length;
-                ingressSizeBytes = ingressSizeBytes + bodySizeBytes + CalculateDictionarySizeBytes(e.Properties) + CalculateDictionarySizeBytes(e.SystemProperties);
+                var bodySizeBytes = e.Body.Length;
+                var propSizeBytes = CalculateDictionarySizeBytes(e.Properties);
+                var syspSizeBytes = CalculateDictionarySizeBytes(e.SystemProperties);
+                ingressSizeBytes += bodySizeBytes + propSizeBytes + syspSizeBytes;
             }
 
             var eventStats = new EventStats()
             {
-                TotalEventsProcessedBytes = ingressSizeBytes,
+                TotalEventsProcessedBytes = Convert.ToDouble(ingressSizeBytes),
             };
 
             return Task.FromResult(eventStats);
         }
 
-        private double CalculateDictionarySizeBytes(IDictionary<string, object> dictionary)
+        public static int CalculateDictionarySizeBytes(IEnumerable<KeyValuePair<string, object>> dictionary)
```

Comment:
Will we lose precision by moving from a `double` to an `int`? Should we change this to a `long`?

Reply:
The old code had an implicit conversion from int to double.  The code to the right of the assignment was still treating everything as an int.

The method Encoding.UTF8.GetByteCount returns int as does the Linq method Aggregate when it binds to the underlying delegates.  If we were worried about overflow once all the bytes are added I could see using a long for the variable we are adding everything into but in that case we wouldn't be able to use the Aggregate method.  Since this represents some of the bytes in an event hub message, and the max message size is 1 MB we should be well under the max size of an int.

Double is the signature for the metrics API which is why we do the conversion.